ambry-store/src/main/java/com.github.ambry.store/Log.java

```diff
@@ -73,6 +77,7 @@
       throw new IOException("Could not read from directory: " + dataDir);
     } else {
       initialize(getSegmentsToLoad(segmentFiles), segmentCapacityInBytes);
+      this.isLogSegmented = isExistingLogSegmented();
```

Comment:
Yes, after initialization, this will reflect the current state of the log, instead of the config. Are there other places that need this info?